uaclient/apt.py

```diff
@@ -653,3 +653,42 @@ def update_esm_caches(cfg) -> None:
     # in tests - down the rabbit hole, not worth it
     except apt.cache.FetchFailedException:
         logging.warning("Failed to fetch the ESM Apt Cache")
+
+
+def _get_apt_config():
+    import apt_pkg
+
+    # We need to clear the config values in case another module
+    # has already initiated it
+    for key in apt_pkg.config.keys():
+        apt_pkg.config.clear(key)
+
+    apt_pkg.init_config()
+    return apt_pkg.config
+
+
+def get_apt_config_keys(key_base):
+    apt_cfg = _get_apt_config()
+    return apt_cfg.list(key_base)
```

Comment:
I think this will only return one level of configuration past the key_base - Does this need to search recursively?

Reply:
Maybe I am missing something, but I think we just want to explore the first level of configuration here. For example, we want all the keys for `Unattended-Upgrades::`. But maybe I am missing your point about search recursively

Reply:
I guess my question is: are there only keys like
* `Unattended-Upgrades::OptionA`
* `Unattended-Upgrades::OptionB`

Or are there also nested keys like:
* `Unattended-Upgrades::Nested::OptionC`
* `Unattended-Upgrades::Nested::OptionD`

and if there are nested keys, does this function need to be recursive in some way to find them?

But if there is only one level of keys that we care about then I think this is okay

Reply:
Ohhh now I see your point. Yes, the function will indeed just return the first level of keys. 

I have checked the [documentation](https://github.com/mvo5/unattended-upgrades/blob/master/README.md#supported-options-reference) for unattended-upgrades to see if we have a nested variable but it doesn't seem like it.

However, since this document doesn't show all of the unattended-upgrade options, I have also checked the [module](https://github.com/mvo5/unattended-upgrades/blob/8c62c05444ed505c2ca96a8092ee6be027033af3/unattended-upgrade) directly and I couldn't find any as well.

So it seems safe for us to just get the top level keys here

Reply:
Sounds good to me!